### PR TITLE
refactor(xy): axis title and related simplifications

### PR DESCRIPTION
**Pull request by monfera**
## Summary

- safer text measurement, no need for manual `destroy`, no need for class
- switched away from the svg measurer in tests and removed both DOM based measurers
- [].sort predicate `compareByValueAsc` equipped to deal with duplicates
- simplified and split `getSimplePadding`
- simpler `computeAxesSizes`, `renderBars` etc.
- removal of remaining shadowing `(ctx) =>` cases
- unified horizontal and vertical titles, due to much shared code
- unified global and panel titles, due to much shared code
- removed parallel redundant switch/case structures etc. from `values.bar.ts`
- replaced two other versions of `clamp` with `clamp`
- removed the git commit and husky
- minor: `forEach` where a `for` is essentially that, etc.

Prepares the ground for https://github.com/elastic/elastic-charts/issues/1310 and https://github.com/elastic/elastic-charts/issues/758 

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [x] All related issues have been linked (i.e. `closes #123`, `fixes #123`)

### CODE REVIEW

File: packages/charts/src/chart_types/xy_chart/rendering/bars.ts [hunk 1]
```diff
@@ -38,163 +38,161 @@ export function renderBars(
...
-    }
-
-    if (absMinHeight !== undefined && height !== 0 && Math.abs(height) < absMinHeight) {
-      const heightDelta = absMinHeight - Math.abs(height);
-      if (height < 0) {
-        height = -absMinHeight;
-        y += heightDelta;
+
+      let y: number | null;
+      let y0Scaled;
```
Comment (monfera) [inline packages/charts/src/chart_types/xy_chart/rendering/bars.ts:54]: Just for the record, I didn't write these `let`s, git lost track of what's happening, still I plan to fix them at some point

File: .huskyrc.js [hunk 2]
```diff
@@ -1,5 +1,3 @@
 module.exports = {
-  hooks: {
-    'commit-msg': 'commitlint -E HUSKY_GIT_PARAMS && yarn typecheck:all && yarn pq && lint-staged',
-  },
+  hooks: {},
```
Comment (markov00) [inline .huskyrc.js:2]: We can completely get rid of the file and remove also the `husky` dependency

File: packages/charts/src/state/selectors/get_legend_size.ts [hunk 3]
```diff
@@ -25,6 +24,8 @@ const MARKER_WIDTH = 16;
 const SHARED_MARGIN = 4;
 const VERTICAL_PADDING = 4;
 const TOP_MARGIN = 2;
+const MAGIC_FONT_FAMILY =
+  '"Inter UI", -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"';
```
Comment (markov00) [inline packages/charts/src/state/selectors/get_legend_size.ts:28]: 🤦 We should find a solution here.
We probably have to wait until EUI has completely switched to CSS-in-JS so we can easily pick the right font from them

File: packages/charts/src/utils/bbox/canvas_text_bbox_calculator.ts [hunk 4]
```diff
@@ -6,44 +6,37 @@
...
+/** @internal */
+export const withTextMeasure = <T>(fun: (textMeasure: TextMeasure) => T) => {
+  const canvas = document.createElement('canvas');
+  canvas.style.display = 'none';
+  const ctx = canvas.getContext('2d');
+  const root = document.documentElement;
+  root.appendChild(canvas);
+  const textMeasure: TextMeasure = ctx
+    ? (text: string, padding: number, fontSize = 16, fontFamily = 'Arial', lineHeight = 1, fontWeight = 400) => {
+        ctx.font = `${fontWeight} ${fontSize}px ${fontFamily}`;
```
Comment (markov00) [inline packages/charts/src/utils/bbox/canvas_text_bbox_calculator.ts:34]: you can use the function `cssFontShorthand` here

Comment (monfera) [reply packages/charts/src/utils/bbox/canvas_text_bbox_calculator.ts:34]: good point! though it constructs than destructs an object for very little gain here, and they need the full Font spec (though it's a pro argument as well)

File: packages/charts/src/state/selectors/get_legend_size.ts (hunk 3 above)
Comment (monfera) [reply packages/charts/src/state/selectors/get_legend_size.ts:28]: I'll put this and a couple of other feedback elements on our code improvement metaissue, and/or fix a couple of them in this PR

File: packages/charts/src/utils/bbox/canvas_text_bbox_calculator.ts (hunk 4 above)
Comment (monfera) [reply packages/charts/src/utils/bbox/canvas_text_bbox_calculator.ts:34]: referenced in #1303

File: packages/charts/src/state/selectors/get_legend_size.ts (hunk 3 above)
Comment (monfera) [reply packages/charts/src/state/selectors/get_legend_size.ts:28]: referenced in #1303

File: packages/charts/src/utils/bbox/canvas_text_bbox_calculator.ts (hunk 4 above)
Comment (monfera) [reply packages/charts/src/utils/bbox/canvas_text_bbox_calculator.ts:34]: added a point to #1303

Comment (markov00) [reply packages/charts/src/utils/bbox/canvas_text_bbox_calculator.ts:34]: yep, I just prefer use the function so I don't have to remember the order of arguments

File: package.json [hunk 5]
```diff
@@ -130,7 +130,6 @@
     "eslint-plugin-react-hooks": "^4.2.0",
     "eslint-plugin-unicorn": "^25.0.1",
     "html-webpack-plugin": "^4.5.2",
-    "husky": "^4.3.6",
```
Comment (nickofthyme) [inline package.json:133]: 🎉 🥂 🍾

**Review by nickofthyme — APPROVED** [review]
A cursory review, code changes LGTM. Thanks for always pushing code quality!

File: package.json (hunk 5 above)
Comment (monfera) [reply package.json:133]: I've second thoughts, maybe we should retain `yarn pq`?

Comment (monfera) [reply package.json:133]: Though in my case at least, repo reformatting is already triggered automatically when saving a file (any file)

**Pull request merged**